src/msw/textctrl.cpp

```diff
@@ -1114,6 +1114,115 @@ void wxTextCtrl::DoSetValue(const wxString& value, int flags)
     }
 }
 
+#if wxUSE_RICHEDIT
+
+DWORD wxCALLBACK MSWEditStreamOutCallback(DWORD_PTR dwCookie, LPBYTE pbBuff,
+    LONG cb, LONG* WXUNUSED(pcb))
+{
+#if wxUSE_THREADS
+    wxMutexGuiLeaveOrEnter();
+#endif // wxUSE_THREADS
+    MSG msg;
+    while ( ::PeekMessage(&msg, (HWND)0, 0, 0, PM_REMOVE) )
+    {
+        if (msg.message != WM_QUIT)
+        {
+            ::TranslateMessage(&msg);
+            ::DispatchMessage(&msg);
+        }
+    }
+
+    // write the text
+    std::string* psEntry = reinterpret_cast<std::string*>(dwCookie);
+    if ( psEntry != nullptr )
+    {
+        psEntry->append(reinterpret_cast<const char*>(pbBuff), cb);
+    }
+
+    return 0;
+}
+
+wxString wxTextCtrl::GetRTFValue() const
+{
+    if ( !IsRich() )
+    {
+        wxFAIL_MSG("RTF support is only available for rich controls!");
+        return wxEmptyString;
+    }
+
+    // The Rich Edit control must write to a char buffer,
+    // which we will later convert to a Unicode wxString using the current code page.
+    //
+    // We will reserve enough space assuming that the RTF will be twice as large
+    // as the unencoded content.
+    std::string buffer;
+    buffer.reserve(GetLastPosition() * 2);
+
+    // Use a EDITSTREAMCALLBACK to stream text out from the control.
+    EDITSTREAM es{ 0 };
+    es.dwError = 0;
+    es.pfnCallback = MSWEditStreamOutCallback;
+    // our callback will write to a char buffer (i.e., std::string)
+    es.dwCookie = reinterpret_cast<DWORD_PTR>(&buffer);
+
+    // Do NOT call with "(CP_UTF8 << 16) | SF_USECODEPAGE", as this will format the
+    // output RTF to UTF-8 with a "urtf1" header (note the 'u').
+    // Although this is a more modern format, few programs
+    // (including Microsoft Wordpad) recognize it.
+    //
+    // Instead, write the content to an ANSI string using the current code page
+    // (which will use a "rtf1" header). The control will encode any Unicode values
+    // outside of the current code page with "\'[0-9A-F][0-9A-F]" syntax that most
+    // parsers would support.
+    //
+    // Finally, use SFF_PLAINRTF so that the generated RTF is more portable between
+    // RTF readers.
+    ::SendMessage(GetHwnd(), EM_STREAMOUT,
+        (WPARAM)(SF_RTF | SFF_PLAINRTF), (LPARAM)&es);
+    return wxString(buffer.c_str(), *wxConvCurrent);
+}
+
+void wxTextCtrl::SetRTFValue(const wxString& val)
+{
+    if ( !IsRich() )
+    {
+        wxFAIL_MSG("RTF support is only available for rich controls!");
+        return;
+    }
```

Comment:
Minor, but this is typically written on a single line using `wxCHECK_RET()`.

Reply:
Got it.